discovery-provider/src/utils/ipfs_lib.py

```diff
@@ -168,41 +168,57 @@ def cat(self, multihash):
             raise  # error is of type ipfshttpclient.exceptions.TimeoutError
 
     def multihash_is_directory(self, multihash):
+        # First attempt to cat multihash locally.
         try:
-            # attempt to cat single byte from CID to determine if dir or file
+            # If cat successful, multihash is not directory.
             self._api.cat(multihash, 0, 1)
             return False
         except Exception as e:  # pylint: disable=W0703
             if "this dag node is a directory" in str(e):
-                logger.warning(f'IPFSCLIENT | Found directory {multihash}')
+                logger.warning(f"IPFSCLIENT | Found directory {multihash}")
                 return True
 
-        # Attempt to retrieve from cnode gateway endpoints
+        # Attempt to retrieve from cnode gateway endpoints.
         gateway_endpoints = self._cnode_endpoints
         for address in gateway_endpoints:
-            gateway_query_address = "%s/ipfs/%s" % (address, multihash)
+            # First, query as dir.
+            gateway_query_address = "%s/ipfs/%s/150x150.jpg" % (address, multihash)
```

Comment:
we should construct these urls properly using urllib

```py
from urllib.parse import urljoin
urljoin(address, f'/ipfs/{multihash}/150x150.jpg')
```

Reply:
good catch

Reply:
good catch, fixed